File: backend/schemas/compra.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime
from decimal import Decimal

class CompraProductoCreate(BaseModel):
    id_producto: int
    cantidad_compra: int
    precio_compra: Decimal


class CompraCreate(BaseModel):
    nit_proveedor: str
    productos: List[CompraProductoCreate]


class CompraGet(BaseModel):
    id_compra: int
    fecha: datetime
    total: Decimal
    nit_proveedor: str


class CompraCabeceraListaOut(BaseModel):
    id_compra: int
    fecha: datetime
    total: Decimal
    nit_proveedor: str
    proveedor_nombre: str


class CompraLineaOut(BaseModel):
    id_producto: int
    cantidad_compra: int
    precio_compra: Decimal
    nombre: Optional[str] = None


class CompraDetalleOut(BaseModel):
    compra: CompraCabeceraListaOut
    lineas: List[CompraLineaOut]

File: frontend/src/types/compra.ts
export type CompraListaItem = {
  id_compra: number;
  fecha: string;
  total: number | string;
  nit_proveedor: string;
  proveedor_nombre: string;
};

export type CompraLineaDetalle = {
  id_producto: number;
  cantidad_compra: number;
  precio_compra: number | string;
  nombre?: string | null;
};

export type CompraDetalleRespuesta = {
  compra: {
    id_compra: number;
    fecha: string;
    total: number | string;
    nit_proveedor: string;
    proveedor_nombre: string;
  };
  lineas: CompraLineaDetalle[];
};
